bgpd: do not allocate lists on fs entries of bgp entries.

pguibert6WIND opened the pull request:
bgp entries in bgp_extra_path structure will be allocated as lists, only
when needed, that is to say when bgp fs entries will be received and
installed on the underlying system.

Signed-off-by: Philippe Guibert <philippe.guibert@6wind.com>

donaldsharp commented:
so 1.5-2 years ago I went through and removed all the list create and deletes when empty code from bgp because of all the bugs that I kept having to fix.  This is in the extra pointer is this what we really want to do?

pguibert6WIND commented:
extra 15% of memory is not only because of that.
there is also the extra structure room space that is getting increased each time.

by the way, I also think of one thing that Lou should check.
I am not sure if VNC calls bgp_extra_info_free() at the good time.

louberger commented:
@donaldsharp are you okay with this being merged?

pguibert6WIND commented:
@louberger , perhaps there are some non related commits.
please tell me what suits you best: all or only the API listnode_add_force()

louberger commented:
this is really more a question for @donaldsharp I was just capturing minutes from the meeting

pguibert6WIND commented:
updated the commits. I added 2 wrappers: one for list_new() and one for listnode_add().
wait for Donald, then for the naming convention + change proposed.

more needed?

louberger commented:
@donaldsharp what do you think?

riw777 commented:
@donaldsharp @louberger  -- these changes look fine to me at this point -- are we waiting on something to push it?

louberger commented:
I just approved it and would like to see it pushed ASAP.

On 4/16/2019 10:14 AM, Russ White wrote:
>
> @donaldsharp <https://github.com/donaldsharp> @louberger 
> <https://github.com/louberger> -- these changes look fine to me at 
> this point -- are we waiting on something to push it?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/FRRouting/frr/pull/3762#issuecomment-483677420>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AGRWyGsqWMMP1TBFBPRO4t4IHqGO-SCoks5vhdqsgaJpZM4avwnj>.
>
